internal/resolution/variable_sources/olm/olm_test.go

```diff
@@ -0,0 +1,156 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		packageRequiredVariables := filterVariables[*required_package.RequiredPackageVariable](variables)
+		Expect(packageRequiredVariables).To(HaveLen(2))
+		Expect(packageRequiredVariables[0].Identifier()).To(Equal(deppy.IdentifierFromString("required package prometheus")))
+		Expect(packageRequiredVariables[0].BundleEntities()).To(HaveLen(2))
+		Expect(packageRequiredVariables[1].Identifier()).To(Equal(deppy.IdentifierFromString("required package packageA")))
+		Expect(packageRequiredVariables[1].BundleEntities()).To(HaveLen(1))
+	})
+
+	It("should produce BundleVariables variables", func() {
+		olmVariableSource := olm.NewOLMVariableSource("prometheus", "packageA")
+		variables, err := olmVariableSource.GetVariables(context.Background(), testEntitySource)
+		Expect(err).ToNot(HaveOccurred())
+
+		bundleVariables := filterVariables[*bundles_and_dependencies.BundleVariable](variables)
+		Expect(bundleVariables).To(HaveLen(3))
+		Expect(bundleVariables).To(WithTransform(func(bvars []*bundles_and_dependencies.BundleVariable) []*input.Entity {
+			var out []*input.Entity
+			for _, variable := range bvars {
+				out = append(out, variable.BundleEntity().Entity)
+			}
+			return out
+		}, Equal([]*input.Entity{
+			entityFromCache("operatorhub/prometheus/0.47.0"),
+			entityFromCache("operatorhub/prometheus/0.37.0"),
+			entityFromCache("operatorhub/packageA/2.0.0"),
+		})))
+	})
+
+	It("should produce GlobalConstraints variables", func() {
+		olmVariableSource := olm.NewOLMVariableSource("prometheus", "packageA")
+		variables, err := olmVariableSource.GetVariables(context.Background(), testEntitySource)
+		Expect(err).ToNot(HaveOccurred())
+
+		globalConstraintsVariables := filterVariables[*global_constraints.GlobalConstraintsVariable](variables)
+		Expect(globalConstraintsVariables).To(HaveLen(6))
+
+		// check global variables have the right names
+		Expect(globalConstraintsVariables).To(WithTransform(func(gvars []*global_constraints.GlobalConstraintsVariable) []string {
+			var out []string
+			for _, variable := range gvars {
+				out = append(out, string(variable.Identifier()))
+			}
+			sort.SliceStable(out, func(i, j int) bool {
+				return strings.Compare(out[i], out[j]) < 0
+			})
+			return out
+		}, Equal([]string{
+			"group:\"foo.io\" version:\"v1\" kind:\"Foo\" gvk uniqueness",
+			"group:\"monitoring.coreos.com\" version:\"v1\" kind:\"Alertmanager\" gvk uniqueness",
+			"group:\"monitoring.coreos.com\" version:\"v1\" kind:\"Prometheus\" gvk uniqueness",
+			"group:\"monitoring.coreos.com\" version:\"v1alpha1\" kind:\"Prometheus\" gvk uniqueness",
+			"packageA package uniqueness",
+			"prometheus package uniqueness",
+		})))
+	})
+
+	It("should return an errors when they occur", func() {
+		olmVariableSource := olm.NewOLMVariableSource("prometheus", "packageA")
+		_, err := olmVariableSource.GetVariables(context.Background(), FailEntitySource{})
+		Expect(err).To(HaveOccurred())
+	})
```

Comment:
Does this sufficiently handle the case where an OLMVariableSource is provided that can't be fulfilled by the entity source?

Reply:
Yeah, it should basically throw it's hand up in the air XD